src/cpp/session/modules/SessionErrors.R

```diff
@@ -77,16 +77,67 @@
    # remove hidden entries from the stack
    stack <- stack[!sapply(stack, is.null)]
 
+   
+   # look for python entry point and fill in the stack from reticulate if we can
+   ammended_stack <- list()
+   lapply(stack, function(x) {
+      func <- x$func
+      if (.rs.hasPythonStackTrace(func)) {
+         foundUserCode <<- TRUE # python code always includes src references
+         python_stack_trace <- .rs.getActivePythonStackTrace()
+         for (item in python_stack_trace)
+            ammended_stack[[length(ammended_stack) + 1]] <<- item
+      } else {
+         ammended_stack[[length(ammended_stack) + 1]] <<- x
+      }
+   })
+   
    # if we found user code (or weren't looking for it), tell the client
    if (foundUserCode || !userOnly)
    {
       err <- list(
-         frames = stack,
+         frames = ammended_stack,
          message = .rs.scalar(geterrmessage()))
       errorReporter(err)
    }
 })
 
+.rs.addFunction("hasPythonStackTrace", function(func) {
+   grepl("^py_call_impl\\(", func) && 
+         .rs.isPackageVersionInstalled("reticulate", "0.7.0.9005") &&
+         !is.null(reticulate::py_last_error())
+})
+
+.rs.addFunction("getActivePythonStackTrace", function() {
+   stack <- list()
+   # silence errors so anything unexpected doesn't mess with global RStudio error handling
+   tryCatch({
+      traceback <- reticulate::py_last_error()$traceback
```

Comment:
I now see the true reason for introducing this `reticulate::py_last_error()` @jjallaire